modules/modelSetup/mixin/ModelSetupDiffusionLossMixin.py

```diff
@@ -53,27 +60,62 @@ def _diffusion_loss(
                     losses = self.align_prop_loss_fn(data['predicted'])
 
             losses = losses * args.align_prop_weight
-        else:
+        else:            
             # TODO: don't disable masked loss functions when has_conditioning_image_input is true.
             #  This breaks if only the VAE is trained, but was loaded from an inpainting checkpoint
             if args.masked_training and not args.model_type.has_conditioning_image_input():
-                losses = loss_util.masked_loss(
-                    F.mse_loss,
-                    data['predicted'],
-                    data['target'],
-                    batch['latent_mask'],
-                    args.unmasked_weight,
-                    args.normalize_masked_area_loss
-                ).mean([1, 2, 3])
+            
+                #MSE/L2 Loss
+                if mse_strength != 0:
+                    losses = loss_util.masked_loss(
+                        F.mse_loss,
+                        data['predicted'],
+                        data['target'],
+                        batch['latent_mask'],
+                        args.unmasked_weight,
+                        args.normalize_masked_area_loss
+                    ).mean([1, 2, 3])
+                    
+                #MAE/L1 Loss
+                if mae_strength != 0:
+                    losses = loss_util.masked_loss(
```

Comment:
Here (and in the mse calculation) you just overwrite the loss instead of adding them up. Later in the calculation, losses is calculated from mse_losses and mae_losses. I think these variables should be used here.